File: Codigo Java/src/Factura.java

package gestorsupermercado;


import java.util.ArrayList;
import java.util.List;

public class Factura {
     private List<String> productos;
    private double total;

    public Factura() {
        productos = new ArrayList<>();
        total = 0.0;
    }

    public void agregarProducto(String nombre, double precio) {
        productos.add(nombre + " - $" + precio);
        total += precio;
    }

    public String generarFactura() {
        StringBuilder sb = new StringBuilder();
        for (String producto : productos) {
            sb.append(producto).append("\n");
        }
        sb.append("Total: $").append(total);
        return sb.toString();
    }
}

File: Codigo/Factura.py
class Factura:
    def __init__(self):
        self.productos = []  
        self.total = 0.0     
    
    def agregar_producto(self, nombre, precio):
        self.productos.append((nombre, precio))  
        self.total += precio                    
    
    def generar_factura(self):
        factura = f"FACTURA\n"
        
        for nombre, precio in self.productos:
            factura += f"- {nombre}: ${precio:.2f}\n"
        
        factura += f"\nTotal: ${self.total:.2f}"
        
        return factura
